### ISSUE TITLE
Wrong expected arg type for compact: Can accept array(s) of variable names

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#267]
This may be related to recent specialized changes for inferences on array functions.

Expected: no error
Observed:

```
ERROR: InvalidArgument - src/zcompact.php:5:20 - Argument 1 of compact expects string, array<int, string> provided     
    return compact(['x']);   
```

```php
<?php

function test() : array {
    $x = 'value';
    return compact(['x']);
}
```

http://php.net/compact

> compact() takes a variable number of parameters. Each parameter can be either a string containing the name of the variable, or an array of variable names.

**Comment by muglug** [comment]
Given it also says

> The array can contain other arrays of variable names inside it; compact() handles it recursively.

I've reverted the signature to `'compact' => ['array', '...var_names='=>'mixed'],`
